core/templates/dev/head/domain/collection/CollectionRightsBackendApiService.js

```diff
@@ -32,12 +54,30 @@ oppia.factory('CollectionRightsBackendApiService', [
           version: collectionVersion,
           is_public: isPublic
         };
-        $http.put(collectionRightsUrl, putParams).then(function() {
+        $http.put(collectionRightsUrl, putParams).then(function(response) {
           // TODO(bhenning): Consolidate the backend rights domain objects and
           // implement a frontend activity rights domain object. The rights
-          // being passed in here should be used to create one of those objects.
-          if (successCallback) {
-            successCallback();
+          // being passed in here should be used to create one of those objects
+
+          // Check if the reponse from the backend does not contradict
+          // putParams.
+          if (response.data.is_private === isPublic) {
```

Comment:
If the backend raises an exception, it should go to the errorResponse clause right?